manifests/passwd_check.pp

```diff
@@ -0,0 +1,32 @@
+# @summary Verify that passwd services (e.g., sssd) are working.
+# The primary use case is to ensure that sssd is configured
+# and running prior to starting slurmctld so we do not lose
+# reservations or prevent users from submitting jobs (both of
+# which require resolution of usernames and potentially groups).
+#
+# Verify that passwd services (e.g., sssd) are working.
+#
+# @param check_users
+#   Users to check for to verify functional passwd services.
+#   CURRENTLY ONLY SUPPORTS AN ARRAY OF ONE USER!
+#
+# @example
+#   include profile_slurm::passwd_check
+class profile_slurm::passwd_check (
+
+  Array[String, 1, 1] $check_users,
+
+) {
+  # if the passwd_check_receipt file doesn't exist, run
+  # the exec_passwd_check, and if that succeeds write the
+  # receipt file;
+  # eventually change this to exec a template that checks
+  # multiple users from check_users (and potentially groups);
+  $passwd_check_user = $check_users[0]
+  $passwd_check_receipt = '/etc/slurm/passwd_check_success'
+  exec { 'exec_passwd_check':
```

Comment:
My gut would have been to instead test if an LDAP group is present. It seems less likely that groups would change in a cluster as much as individual users might.

Reply:
That's a good idea, I will change it to check a group.